pkgs/development/libraries/phonon/backends/gstreamer.nix

```diff
@@ -29,6 +29,23 @@ stdenv.mkDerivation rec {
     sha256 = "1wc5p1rqglf0n1avp55s50k7fjdzdrhg0gind15k8796w7nfbhyf";
   };
 
+  # Hardcode paths to useful plugins so the backend doesn't depend
+  # on system paths being set.
+  patches = [ ./gst-plugin-paths.patch ];
+  gstPluginPaths =
+    lib.makeSearchPathOutput "lib" "/lib/gstreamer-1.0"
+    (with gst_all_1; [
+      gstreamer
+      gst-plugins-base
+      gst-plugins-good
+      gst-plugins-ugly
+      gst-plugins-bad
+      gst-libav
+    ]);
+  postPatch = ''
+    substituteInPlace gstreamer/backend.cpp --subst-var gstPluginPaths
```

Comment:
Since the patch file is so small and it relies on a generated value from Nix, why not inline it and template the `gstPluginPaths`? That way you don't need the `substituteInPlace` step.